### DISCUSSION TITLE
RSSI and LQ in Betaflight

### DISCUSSION THREAD

**Discussion by richardbulger** [ExpressLRS/ExpressLRS#692] [Q&A]
From the [wiki entry regarding the betaflight setup of RSSI and LQ](https://github.com/ExpressLRS/ExpressLRS/wiki/OpenTX-and-Betaflight-Setup#rssi-and-link-quality):

> To get RSSI and Link Quality displayed in the OSD, set RSSI Channel to "Disabled" in the Receiver tab of the Betaflight Configurator, and RSSI_ADC should be disabled on the Configuration tab. Both of these are the default. On the OSD menu, use the "Link Quality" and "RSSI dBm value" elements (not "RSSI Value").

If I follow these instructions, I get 0 LQ 0 RSSI and -150 dbm RSSI.
The only thing that worked for me is setting the RSSI channel to 11, and using RSSI (not "RSSI dBm value") in the OSD.

Am I doing something wrong or does the wiki need updating?

**Comment by JBKingdon** [comment]
The wiki looks right for betaflight - is that what you are running? Are you using a normal MAX7456 OSD or something different? DJI is it's own thing for example, with particular requirements for RSSI.

**Reply by richardbulger** [reply]
I am using the Mamba f405 stack, with DJI

**Reply by JBKingdon** [reply]
ah, ok, so that's normal for DJI. There's a (difficult to find with search :disappointed: ) section for DJI in the wiki page for setting up BF https://github.com/ExpressLRS/ExpressLRS/wiki/OpenTX-and-Betaflight-Setup#rssi-and-link-quality

> If using DJI Goggles, you're required to use "RSSI Value" as the OSD element. Therefore you have to decide between LQ or RSSI, by selecting either AUX11 (LQ) or AUX12 (RSSI) as RSSI Channel on the Receiver tab.

**Reply by richardbulger** [reply]
That's the same page I linked :) However, that sentence implies that you can still choose between LQ or RSSI. I would not get any signal with AUX12 (RSSI) selected

**Reply by JBKingdon** [reply]
That seems odd - I can't think of any reason AUX12 wouldn't work if AUX11 does. You can check that values are being sent on both channels in the receiver tab of the configurator.

**Reply by theikalman** [reply]
I have the same stack, mamba f405 MK2, but with analog. The same issue happen to me, mine I can select `AUX11/AUX12` as RSSI channel in receiver tab, and it is shown correctly in OSD (RSSI), however the LQ is always 0.

I am able to sorted it out by enable RSSI_ADC in configuration tab, save, and then disable RSSI_ADC in configuration tab again (also RSSI channel in receiver tab should be disabled at this point).

I am not quite sure what the cause is but it fix the issue and now the LQ and RSSI/RSSI dbm in OSD displayed correctly.

**Comment by ghost** [comment]
The wiki is correct. The wiki deals with analog OSD since BF can pull LQ and RSSI-DBM directly without it needing to be added through a channel. For DJI (which doesn't support RSSI-DBM and LQ), you have to pass it through the RSSI channel and enable RSSI in the OSD. Works as stated in the wiki.

**Reply by ofirattia** [reply]
The part that refer to set the rssi source via the CLU is only for iNav?
